glib/src/subclass/types.rs

```diff
@@ -95,7 +95,7 @@ unsafe impl<T: InstanceStruct> InstanceStructExt for T {
             let data = Self::Type::type_data();
             let private_offset = data.as_ref().impl_offset();
             let ptr: *const u8 = self as *const _ as *const u8;
-            let imp_ptr = ptr.offset(private_offset);
+            let imp_ptr = ((ptr as isize) + private_offset) as *const u8;
```

Comment:
Could you add comments explaining why we do it like and also a note saying that once `expose_addr` is stable, we should use it instead please?

Reply:
Already done locally